projects/dsp-ui/src/lib/action/pipes/string-transformation/truncate.pipe.ts

```diff
@@ -25,14 +24,18 @@ import { Pipe, PipeTransform } from '@angular/core';
 })
 export class TruncatePipe implements PipeTransform {
 
-    transform(value: string, args: string[]): string {
+    transform(value: string, limit?: number, trail?: string): string {
 
-        if (!value || value.length === 0) {
-            return;
+        if (typeof value !== 'string' || value.length === 0) {
+            return '';
         }
 
-        const limit = args.length > 0 ? parseInt(args[0], 10) : 20;
-        const trail = args.length > 1 ? args[1] : '...';
+        // default to 20 if no character limit is provided
+        if (!limit) { limit = 20; }
```

Comment:
I think it is not a good practice to reassign to a parameter (https://eslint.org/docs/rules/no-param-reassign).

I think you could do the following:

```ts
export class TruncatePipe implements PipeTransform {

    defaultLimit = 20;
    defaultTrail = '...';

    transform(value: string, limit?: number, trail?: string): string {

        if (typeof value !== 'string' || value.length === 0) {
            return '';
        }

        const limitSetting: number = limit !== undefined ? limit : this.defaultLimit;
        const trailSetting: string = trail !== undefined ? trail : this.defaultTrail;

        return value.length > limitSetting ? value.substring(0, limitSetting) + trailSetting : value;
    }

}
```

The ternary operators are not easy to understand but quite elegant. Maybe just add some comments?

Reply:
Ha, yeah that was a rookie move on my part. I should know better 😅

Reply:
rookie mistake has been fixed in e5491d7